rpc/utils.go

```diff
@@ -218,29 +218,27 @@ func newSubscriptionId() (string, error) {
 // SupportedModules returns the collection of API's that the RPC server offers
 // on which the given client connects.
 func SupportedModules(client Client) (map[string]string, error) {
-	req := map[string]interface{}{
-		"id":     1,
-		"method": "rpc_modules",
+	req := JSONRequest{
+		Id:      new(int64),
+		Version: "2.0",
+		Method:  "rpc_modules",
 	}
-
 	if err := client.Send(req); err != nil {
 		return nil, err
 	}
 
-	var response map[string]interface{}
-	if err := client.Recv(&response); err != nil {
+	response := new(JSONSuccessResponse)
```

Comment:
Not if you force `new` and exactly because of Go's escape analysis you don't need to use `new`, it will know what to do. Best to leave it therefor as a stack variable and leave it up to the compiler to determine whether it needs to be moved to the heap or not.